### ISSUE TITLE
Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu

### ISSUE THREAD

**Issue by chdelacr** [comfyanonymous/ComfyUI#2363]
Happens when loading SVD model, other models aren't affected.

Output from console:

> Error occurred when executing KSampler:
> 
> Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
> 
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\execution.py", line 153, in recursive_execute
> output_data, output_ui = get_output_data(obj, input_data_all)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\execution.py", line 83, in get_output_data
> return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\execution.py", line 76, in map_node_over_list
> results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1299, in sample
> return common_ksampler(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1269, in common_ksampler
> samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\sample.py", line 101, in sample
> samples = sampler.sample(noise, positive_copy, negative_copy, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 716, in sample
> return sample(self.model, noise, positive, negative, cfg, self.device, sampler, sigmas, self.model_options, latent_image=latent_image, denoise_mask=denoise_mask, callback=callback, disable_pbar=disable_pbar, seed=seed)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 622, in sample
> samples = sampler.sample(model_wrap, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 561, in sample
> samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
> return func(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\k_diffusion\sampling.py", line 137, in sample_euler
> denoised = model(x, sigma_hat * s_in, **extra_args)
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
> return self._call_impl(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
> return forward_call(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 285, in forward
> out = self.inner_model(x, sigma, cond=cond, uncond=uncond, cond_scale=cond_scale, model_options=model_options, seed=seed)
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
> return self._call_impl(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
> return forward_call(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 275, in forward
> return self.apply_model(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 272, in apply_model
> out = sampling_function(self.inner_model, x, timestep, uncond, cond, cond_scale, model_options=model_options, seed=seed)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 252, in sampling_function
> cond_pred, uncond_pred = calc_cond_uncond_batch(model, cond, uncond_, x, timestep, model_options)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 226, in calc_cond_uncond_batch
> output = model.apply_model(input_x, timestep_, **c).chunk(batch_chunks)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\model_base.py", line 85, in apply_model
> model_output = self.diffusion_model(xc, t, context=context, control=control, transformer_options=transformer_options, **extra_conds).float()
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
> return self._call_impl(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
> return forward_call(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\diffusionmodules\openaimodel.py", line 854, in forward
> h = forward_timestep_embed(module, h, emb, context, transformer_options, time_context=time_context, num_video_frames=num_video_frames, image_only_indicator=image_only_indicator)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\diffusionmodules\openaimodel.py", line 38, in forward_timestep_embed
> x = layer(x, emb, num_video_frames, image_only_indicator)
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
> return self._call_impl(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
> return forward_call(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\diffusionmodules\openaimodel.py", line 343, in forward
> x = self.time_mixer(
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
> return self._call_impl(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
> return forward_call(*args, **kwargs)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\diffusionmodules\util.py", line 79, in forward
> alpha = self.get_alpha(image_only_indicator)
> File "D:\Programas\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\diffusionmodules\util.py", line 61, in get_alpha
> alpha = torch.where(

**Comment by Lilneo786** [comment]
```
model.to(device)  # Move the model to the desired device
input_data = input_data.to(device)  # Move input data to the same device

```

`device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")`

**Comment by chdelacr** [comment]
> ```
> model.to(device)  # Move the model to the desired device
> input_data = input_data.to(device)  # Move input data to the same device
> ```
> 
> `device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")`

thanks but could you please elaborate?

**Comment by comfyanonymous** [comment]
Should be fixed now.

**Comment by chdelacr** [comment]
> Should be fixed now.

Cool, thanks a lot!
